woodwork/table_accessor.py

```diff
@@ -170,6 +171,27 @@ def __getitem__(self, key):
         series.ww.init(**column, use_standard_tags=self.use_standard_tags)
         return series
 
+    def __setitem__(self, col_name, column):
+        series = tuple(pkg.Series for pkg in (pd, dd, ks) if pkg)
+        if not isinstance(column, series):
+            raise ValueError('New column must be of Series type')
+
+        # Don't allow reassigning of index or time index with setitem
+        if self.index == col_name:
+            raise KeyError('Cannot reassign index. Change column name and then use df.ww.set_index to reassign index.')
+        if self.time_index == col_name:
+            raise KeyError('Cannot reassign time index. Change column name and then use df.ww.set_time_index to reassign time index.')
+
+        if column.name is not None and column.name != col_name:
+            warnings.warn(ColumnNameMismatchWarning().get_warning_message(column.name, col_name),
+                          ColumnNameMismatchWarning)
+
+        if column.ww._schema is None:
+            column.ww.init(use_standard_tags=self.use_standard_tags)
```

Comment:
Here, we want to initialize Woodwork for the column if it hasn't been already initialized. If a user passes a column that fails the init call, wouldn't the column-related error be raised from that init call and tell the user how to fix?

I'll switch to `init_series`. Is there a better way to check if a column has been initialized other than `column.ww._schema is None`?

Reply:
I think the issue is that the error that is raised when a column fails init, is very specific to initialization: https://github.com/alteryx/woodwork/blob/42e57bc5afd971ff47e66f4abbb1065eb5d6c603/woodwork/column_accessor.py#L225

However, in this case the user isn't calling `init` we are doing it for them behind the scenes, so I was thinking it could be confusing for a user to see a message about initializing when they never called init explicitly. You are correct, that the message the would see does contain information on how to fix though.

Unless I'm overlooking something I think you should still be able to use `init_series` inside the current check of `if column.ww._schema is None:`.

Reply:
Maybe we should consider making the error more generic so it can be used here as well and make sense in this context?

Reply:
Another option would be to only allow users to assign a column that has already been initialized, which would force the user to handle initialization on their own, either with `df.ww.init` or `init_series`.

Reply:
It'd be worth noting in the docs

Is there another way to check if Woodwork has been initialized in the column? I used `column.ww._schema is None` to check but not sure there's anything more direct (`column.ww.is_initialized`)?

Reply:
The schema check should be fine for initialization, and I don't think there is anything else to check at this point. If the schema exists Woodwork has been initialized successfully, otherwise the schema will be none, so I think this should be a pretty reliable check.

Reply:
I think the error is helpful in this context. I like the idea of making it more generic.

Reply:
✅  8f5317e